ext/drx/scatter_gather_aarch64.c

```diff
@@ -630,10 +653,10 @@ expand_contiguous(void *drcontext, instrlist_t *bb, instr_t *sg_instr,
      * time.
      */
 
-    /* index    scratch_vec.element_size, #0, #1 */
+    /* index    scratch_vec.element_size, #0, #reg_count */
     EMIT(index_sve, opnd_create_reg_element_vector(scratch_vec, sg_info->element_size),
```

Comment:
So for ld4b, the first vector gets the first byte of all elements, second vector gets the second byte of all elements and so on till the fourth vector (where each element is 4 byte).

Curious to know a use case where one'd want to load bytes like that. Do you have an example?

Reply:
It can be used to load an array of struct-like data. The struct gets deinterleaved so each register gets the values of a particular member.

```
struct Rgba {
    uint8_t r;
    uint8_t g;
    uint8_t b;
    uint8_t a;
};

A rgba[num];
```

Load with:
```ld4b {z0, z1, z2, z3}, pg/z [rgba]```

Now `z0` has all `r` values, `z1` has all `g` values, `z2` has all `b` values, `z3` has all `a` values.

Reply:
Thanks.